crates/router/src/encryption.rs

```diff
@@ -1,38 +1,116 @@
-use crate::{
-    errors, headers, services,
-    types::{domain::EncryptionCreateRequest, RequestContent, Response},
-    SessionState,
-};
+use std::str::FromStr;
+
+use error_stack::ResultExt;
+use http::{HeaderMap, HeaderName, HeaderValue};
+#[cfg(feature = "keymanager_mtls")]
+use masking::PeekInterface;
+use once_cell::sync::OnceCell;
+
+use crate::{errors, headers, types::domain::EncryptionCreateRequest, SessionState};
+
+static ENCRYPTION_API_CLIENT: OnceCell<reqwest::Client> = OnceCell::new();
+
+#[allow(unused_mut)]
+pub fn get_api_encryption_client(
+    state: &SessionState,
+) -> errors::CustomResult<reqwest::Client, errors::ApiClientError> {
+    let proxy = &state.conf.proxy;
+
+    let get_client = || {
+        let mut client = reqwest::Client::builder()
+            .redirect(reqwest::redirect::Policy::none())
+            .pool_idle_timeout(std::time::Duration::from_secs(
+                proxy.idle_pool_connection_timeout.unwrap_or_default(),
+            ));
+
+        #[cfg(feature = "keymanager_mtls")]
+        {
+            let ca = state.conf.key_manager.ca.clone();
+            let cert = state.conf.key_manager.cert.clone();
+
+            let identity = reqwest::Identity::from_pem(cert.peek().as_ref())
+                .change_context(errors::ApiClientError::ClientConstructionFailed)?;
+            let ca_cert = reqwest::Certificate::from_pem(ca.peek().as_ref())
+                .change_context(errors::ApiClientError::ClientConstructionFailed)?;
```

Comment:
you can use 
```rust
if cfg!(feature = "keymanager_mtls") 
```

you would not need #[allow(unused_mut)] in that case

Reply:
We cannot do this since ca and cert have compile time feature constraints